dataikuapi/dssclient.py

```diff
@@ -626,8 +634,73 @@ def create_global_api_key(self, label=None, description=None, admin=False):
             raise Exception('API key creation failed : %s' % (json.dumps(resp.get('messages', {}).get('messages', {}))))
         if not resp.get('id', False):
             raise Exception('API key creation returned no key')
-        key = resp.get('key', '')
-        return DSSGlobalApiKey(self, key)
+
+        if as_type == 'object':
+            return DSSGlobalApiKey(self, resp.get('id', ''))
+        elif as_type == 'dict':
+            return resp
+        else:
+            raise ValueError("Unknown as_type")
+    ########################################################
+    # Personal API Keys
+    ########################################################
+
+    def list_personal_api_keys(self, as_type='dict'):
+        """
+        List all personal API keys:
+            - not admin: only the keys belonging to the owner of the key
+            - admin: All the personal keys
+
+        :param as_type: 'objects' or 'dict'
+
+        :returns: All personal API keys, as a list of dicts
+        """
+
+        resp = self._perform_json(
+            "GET", "/personal-api-keys/")
+        if as_type == 'objects':
+            return [DSSPersonalApiKey(self, item.id) for item in resp]
+        elif as_type == 'dict':
+            return resp
+        else:
+            raise ValueError("Unknown as_type")
+
+    def get_personal_api_key(self, id):
+        """
+        Get a specific API key from the user doing the request
+
+        :param str id: the id the desired API key
+
+        :returns: A :class:`dataikuapi.dss.admin.DSSPersonalApiKey` API key handle
+        """
+        return DSSPersonalApiKey(self, id)
+
+    def create_personal_api_key(self, on_behalf="", label="", description="", as_type='object'):
```

Comment:
There were a comment from @mhoarau on the previous PR about backward compatibility, in this case shouldn't `label` be the second argument, and `as_type` default to `dict`? (sorry if I missed the resolution of that comment)

Reply:
I think your comment was meant for the method `create_global_api_key`, as `create_personal_api_key` is actually new.